CHANGELOG.md

```diff
@@ -1,5 +1,7 @@
 ## Unreleased
- * r/tfe_workspace: Changes in `agent_pool_id` and `execution_mode` attributes are now detected and applied. ([#607](https://github.com/hashicorp/terraform-provider-tfe/pull/607))
+
+BREAKING CHANGES:
```

Comment:
If this is a breaking change, is it time for a major version bump? cc @brandonc

Reply:
The provider is still pre-1.0 so it wouldn't be a major version bump

Reply:
Yep! This also wouldn't be the first time we introduce a breaking change, see v0.32.0 or v0.35.0 :heart: